pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -186,12 +185,14 @@ def _zyz_decomposition(U, wire, return_global_phase=False):
 
     phis, thetas, omegas, alphas = map(math.squeeze, [phis, thetas, omegas, alphas])
 
-    phis = phis % (4 * numpy.pi)
-    thetas = thetas % (4 * numpy.pi)
-    omegas = omegas % (4 * numpy.pi)
+    phis = phis % (4 * np.pi)
+    thetas = thetas % (4 * np.pi)
+    omegas = omegas % (4 * np.pi)
 
     operations = [qml.RZ(phis, wire), qml.RY(thetas, wire), qml.RZ(omegas, wire)]
     if return_global_phase:
+        if qml.math.all(np.imag(alphas) < 1e-15):
+            alphas = np.real(alphas)
```

Comment:
and if they're not all below this threshold... isn't something wrong? in other words, should we add an `else: raise ValueError("invalid global phase")`?

Reply:
I've now realized it will never be imaginary - it is from `qml.math.angle(dets)` in `_convert_to_su2`, which is real-valued if `dets` is complex, and 0 if its not.